src/ui/task-tree-stats.tsx

```diff
@@ -0,0 +1,69 @@
+import { TaskManager } from '@jujulego/tasks';
+import { Text } from 'ink';
+import Spinner from 'ink-spinner';
+import symbols from 'log-symbols';
+import { useLayoutEffect, useState } from 'react';
+
+// Types
+export interface TaskTreeStatsProps {
+  readonly manager: TaskManager;
+}
+
+// Component
+export default function TaskTreeStats({ manager }: TaskTreeStatsProps) {
+  // Follow stats
+  const [stats, setStats] = useState(() => {
+    const base = { running: 0, done: 0, failed: 0 };
+    
+    for (const task of manager.tasks) {
+      switch (task.status) {
+        case 'starting':
+        case 'running':
+          base.running += task.weight;
+          break;
+
+        case 'done':
+          base.done += task.weight;
+          break;
+
+        case 'failed':
+          base.failed += task.weight;
+          break;
+      }
+    }
+    
+    return base;
+  });
+
+  useLayoutEffect(() => manager.on('started', (task) => {
+    setStats((old) => ({
+      ...old,
+      running: old.running + task.weight
+    }));
+  }), [manager]);
+
+  useLayoutEffect(() => manager.on('completed', (task) => {
+    setStats((old) => ({
+      running: old.running - task.weight,
+      done: task.status === 'done' ? old.done + task.weight : old.done,
+      failed: task.status === 'failed' ? old.failed + task.weight : old.failed,
+    }));
+  }), [manager]);
+
+  // Render
+  return (
+    <Text>
+      { (stats.running !== 0) && (
+        <><Spinner type={'sand' as 'dots'} /> <Text bold>{ stats.running }</Text> running</>
```

Comment:
Remove cast when https://github.com/sindresorhus/cli-spinners/pull/69 is published